UnityProject/Assets/Scripts/Messages/Client/RequestAuthMessage.cs

```diff
@@ -14,25 +14,43 @@ public class RequestAuthMessage : ClientMessage
 	public override IEnumerator Process()
 	{
 		//	Debug.Log("Processed " + ToString());
-
 		yield return WaitFor(SentBy);
-		Debug.Log("Server Starting Auth for User:" + SteamID);
 
-		if (Server.Instance != null && SteamID != null && TicketBinary != null)
+//		if ( !Managers.instance.isForRelease )
+//		{
+//			Debug.Log($"Ignoring {this}, not for release");
+//			yield break;
+//		}
+		
+		var connectedPlayer = PlayerList.Instance.Get( NetworkObject );
+
+		if ( PlayerList.IsConnWhitelisted(connectedPlayer) )
+		{
+			Debug.Log( $"Player whitelisted: {connectedPlayer}" );
+			yield break;
+		}
+		if ( connectedPlayer.SteamId != 0 )
+		{
+			Debug.Log( $"Player already authenticated: {connectedPlayer}" );
+			yield break;
+		}
+		
+//		Debug.Log("Server Starting Auth for User:" + SteamID);
+		if (Server.Instance != null && SteamID != 0 && TicketBinary != null)
 		{
-			//FIXME Prevent run twice for already verified player
 			
 			//This results in a callback in CustomNetworkManager
 			if (!Server.Instance.Auth.StartSession(TicketBinary, SteamID))
 			{
 				// This can trigger for a lot of reasons
 				// More info: http://projectzomboid.com/modding//net/puppygames/steam/BeginAuthSessionResult.html
 				// if triggered does prevent the authchange callback.
-				Debug.Log("Start Session returned false");
+//				Debug.Log("Start Session returned false, kicking");
```

Comment:
The debug log may come in handy to trace if a player has been kicked by the server